radiography/src/main/java/radiography/Views.kt

```diff
@@ -18,3 +19,8 @@ fun View?.scan(
     Radiography.scan(this, includeTextViewText, textViewTextMaxLength, viewFilter)
   }
 }
+
+internal fun ViewGroup.childrenAsList(): List<View> = object : AbstractList<View>() {
```

Comment:
Need to account for things changing under us mid way. Let's instead create a mutable list and iterate, skipping if null?

Reply:
I'd go with just one mutable list, right where we also do the filtering.

Reply:
How would things change midway, unless they're being done from another thread?

Reply:
The old code didn't make a defensive copy either.

Reply:
Another thread: yep!


E.g. when rendering the view hierarchy in a default exception handler that received a background thread crash.


Prior impl was somewhat defensive by dealing with "maybe I'll get a null child"

Reply:
Ah, that makes sense. But I thought Views threw if accessed from different threads like that?

> Let's instead create a mutable list and iterate, skipping if null?

And I'm not quite clear on what you meant. `List` is a type that indicates a read-only list, not an immutable list, so the contract still allows the list contents to change out from under us. I think we need to make the type argument nullable again though, since there's no way to actually synchronize on the view's children itself or get them all atomically.

Reply:
Sorry that wasn't clear. I meant to do a best effort + safe.

So instead of this:

```kotlin
      override val View.children: Collection<View?>
        get() = if (this is ViewGroup) childrenAsList().filter(viewFilter::matches) else emptyList()
```

Try:

```
      override val View.children: Collection<View?>
        get() = if (this is ViewGroup) {
          val children = mutableListOf<View>
          for (i in 0..childCount) {
            getChildAt(i)?.let { children += it }
          }
          children
        } else emptyList()
```